x-pack/platform/plugins/private/data_visualizer/public/lite/index_input.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFieldText, EuiFormRow, EuiSpacer, EuiTitle } from '@elastic/eui';
+import type { FC } from 'react';
+import React, { useState } from 'react';
+import useDebounce from 'react-use/lib/useDebounce';
+import type { FileUploadStartApi } from '@kbn/file-upload-plugin/public/api';
+import { i18n } from '@kbn/i18n';
+import { STATUS } from './file_manager/file_manager';
+
+interface Props {
+  setIndexName: (name: string) => void;
+  setIndexValidationStatus: (status: STATUS) => void;
+  fileUpload: FileUploadStartApi;
+}
+
+export const IndexInput: FC<Props> = ({ setIndexName, setIndexValidationStatus, fileUpload }) => {
+  const [indexNameLocal, setIndexNameLocal] = useState('');
+  const [indexNameError, setIndexNameError] = useState('');
+
+  useDebounce(
+    async () => {
+      setIndexValidationStatus(STATUS.STARTED);
+      if (indexNameLocal === '') {
+        setIndexValidationStatus(STATUS.COMPLETED);
+        setIndexNameError('');
+        setIndexName('');
+        return;
+      }
+      const exists = await fileUpload.checkIndexExists(indexNameLocal);
```

Comment:
Probably not related to the changes here, but noticed you get this error if the index name you enter ends in a `*`:

<img width="531" alt="Screenshot 2025-01-28 at 12 11 46" src="https://github.com/user-attachments/assets/ed73aade-db6a-4c7c-9f22-8c348d78efc3" />